cola/widgets/stash.py

```diff
@@ -192,7 +216,7 @@ def stash_apply(self, pop=False):
         context = self.context
         index = get(self.keep_index)
         cmds.do(stash.ApplyStash, context, selection, index, pop)
-        QtCore.QTimer.singleShot(1, self.accept)
+        self.update_from_model()
```

Comment:
Just double-checking -- the deferred call to `self.accept` was making sure we close the dialog here.  For "Apply", I can see wanting to apply more than one stash (though it is atypical) so we'd want to keep the dialog open.

But for save below, we also remove that call -- but I think we would want to close the dialog after saving, or do you have a different workflow where you keep the dialog open for multiple actions?

Reply:
You're completely right, such a workflow seems quite… dirty, to say the least. But I would expect the window to only close when I ask it to; the button reads "Save", not "Save and close"… What do you think regarding the UX?

Reply:
It's okay to keep it as "Save" since it matches the expected one-off nature of stashing something.  It's very convenient that we close the dialog, since otherwise the user would be burdened with having to do so for the common case.

Another concern is that we already have translations written for the `Save` string (see the `po/` directory), so sometimes it's limiting but I do try to avoid changing existing strings that already have established translations in place.  I should probably document that in the guidelines so that it's taken into consideration.

I drive it from the keyboard.  The "Save" button is the default action, so to save a stash it's `shift-alt-s enter stash-name enter` which is slick from a keyboard usability perspective.  I don't want to hit `esc` at the end to close the dialog because it encourages a sloppy workflow where the user has tons of stashes.

If we want to apply a different stash after saving, then it's not that inconvenient to hit the `shift-alt-s` shortcut and bring up the dialog again.  That said, we don't have an easy way to apply stash without using the mouse, so I'll try to add that shortly.

If I push that up, apologies if you have to rebase for it.

Reply:
Alrighty, I just pushed up some small changes to the widget that adds hotkeys for apply, pop, and drop.  Please rebase.  The changes were pretty minimal (mostly purely additive) so it should be easy to rebase your tree against it, but let me know if you run into any snags.

k,j navigates up,down the stash list.  Ctrl+Enter applies stashes.  Ctrl+Backspace pops the stash.  Ctrl+Shift+Backspace drops the stash.  Save is the default action so it gets the Enter hotkey by default from Qt.

We can later maybe add the Edit Ctrl+E hotkey for the new rename action here. We're "editing" the stash name, so it might make sense to take over that action/verb in this context.